cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/web/ACSearchRunner.java

```diff
@@ -163,122 +163,102 @@ public void run()
-    {
+    private void search(Subject subject) {
 
         // Note: This search runner is customized to run with
         // InMemoryJobPersistence, and synchronous POST requests are
         // dealt with immediately, rather than returning results via
         // a redirect.
         // Jobs in this runner are never updated after execution begins
-        // in case the in-memory job has gone away.  Error reporting
+        // in case the in-memory job has gone away. Error reporting
         // is done directly through the response on both POST and GET
 
-        try
-        {
-            ExecutionPhase ep =
-                jobUpdater.setPhase(job.getID(), ExecutionPhase.QUEUED,
-                                    ExecutionPhase.EXECUTING, new Date());
-            if ( !ExecutionPhase.EXECUTING.equals(ep) )
-            {
+        try {
+            ExecutionPhase ep = jobUpdater.setPhase(job.getID(), ExecutionPhase.QUEUED, ExecutionPhase.EXECUTING,
+                    new Date());
+            if (!ExecutionPhase.EXECUTING.equals(ep)) {
                 throw new IllegalStateException("QUEUED -> EXECUTING [FAILED]");
             }
             log.debug(job.getID() + ": QUEUED -> EXECUTING [OK]");
 
-            RequestValidator rv = new RequestValidator();
-            rv.validate(job.getParameterList());
-
-            // only allow users to search themselves...
-            //Principal userBeingSearched = rv.getPrincipal();
+            validateParams(job.getParameterList());
 
             PluginFactory factory = new PluginFactory();
             GroupPersistence dao = factory.createGroupPersistence();
             Collection<Group> groups;
-            try
-            {
-                groups = dao.getGroups(rv.getRole(), rv.getGroupID());
-            }
-            catch(GroupNotFoundException ignore)
-            {
+            try {
+                groups = dao.getGroups(role, groupName);
+            } catch (GroupNotFoundException ignore) {
                 log.debug("no memberships found");
                 groups = new ArrayList<Group>();
             }
-            syncOut.setResponseCode(HttpServletResponse.SC_OK);
-            GroupListWriter groupListWriter = new GroupListWriter();
-            groupListWriter.write(groups, syncOut.getOutputStream());
-        }
-        catch (TransientException t)
-        {
+            
+            if (ivoaStandardResponse) {
+                if (groups.size() > 0) {
+                    StringBuilder response = new StringBuilder();
+                    for (Group g : groups) {
+                        response.append(g.getID().getName());
+                        response.append("\n");
+                    }
+                    syncOut.setHeader("Content-Type", "text/plain");
+                    syncOut.getOutputStream().write(response.toString().getBytes());
+                }
+            } else {
+                GroupListWriter groupListWriter = new GroupListWriter();
```

Comment:
Is the legacy XML output temporary? The current hack is OK for now, but we should move to one of:
* param: RESPONSEFORMAT = text/plain or text/xml
* header: Accept: text/xml (default to plain)
for things like this if we need to keep XML around.

Reply:
Okay, I've added a TODO for this